src/Discord.Net.Core/Entities/Interactions/MessageComponents/ComponentBuilder.cs

```diff
@@ -81,6 +81,35 @@ internal void AddComponent(IMessageComponent component, int row)
             }
         }
 
+        /// <summary>
+        ///     Removes all components of the given type from the <see cref="ComponentBuilder"/>.
+        /// </summary>
+        /// <param name="t">The <see cref="ComponentType"/> to remove.</param>
+        /// <returns>The current builder.</returns>
+        public ComponentBuilder RemoveComponentsOfType(ComponentType t)
+        {
+            this.ActionRows.ForEach(ar => ar.Components.RemoveAll(c => c.Type == t));
+            return this;
+        }
+
+        /// <summary>
+        ///     Removes a component from the <see cref="ComponentBuilder"/>.
+        /// </summary>
+        /// <param name="customId">The custom id of the component.</param>
+        /// <returns>The current builder.</returns>
+        public ComponentBuilder RemoveComponent(string customId)
+        {
+            this.ActionRows.ForEach(ar => ar.Components.RemoveAll(c => c.CustomId == customId || (c is ButtonComponent b && b.Url == customId)));
```

Comment:
Why does this remove matching link buttons but not document this behaviour? This may cause unexpected behaviour, being difficult to debug. I would suggest making that an optional feature, or taking it out of this method entirely.